### PR TITLE
chore: Remove some calls to `unwrap`

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Related to https://github.com/apache/datafusion-comet/issues/597

## Rationale for this change

it is better to return `Err` than to call `unwrap`.

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -108,7 +108,7 @@ strip = "debuginfo"
 [lib]
 name = "comet"
 # "rlib" is for benchmarking with criterion.
-crate_type = ["cdylib", "rlib"]
```
Comment (andygrove) [inline core/Cargo.toml:111]: drive-by cleanup to fix a warning since we upgraded to Rust 1.79. Unrelated to other changes in this PR.

**Review by huaxingao — APPROVED** [review]
LGTM. Thanks for the PR!
